xdsl/dialects/vector.py

```diff
@@ -54,4 +54,99 @@ def get(vector: Operation | SSAValue, ref: Operation | SSAValue,
         return Store.build(operands=[vector, ref, indices])
 
 
-Vector = Dialect([Load, Store], [])
+@irdl_op_definition
+class Broadcast(Operation):
+    name = "vector.broadcast"
+    source: Annotated[Operand, AnyAttr()]
+    vector: Annotated[OpResult, VectorType]
+
+    def verify_(self):
+        if self.source.typ != self.vector.typ.element_type:
+            raise VerifyException(
+                "Source operand and result vector must have the same element type."
+            )
+
+    @staticmethod
+    def get(source: Operation | SSAValue) -> Broadcast:
+        return Broadcast.build(operands=[source],
+                               result_types=[
+                                   VectorType.from_type_and_list(
+                                       SSAValue.get(source).typ)
+                               ])
+
+
+@irdl_op_definition
+class FMA(Operation):
+    name = "vector.fma"
+    lhs: Annotated[Operand, VectorType]
+    rhs: Annotated[Operand, VectorType]
+    acc: Annotated[Operand, VectorType]
+    res: Annotated[OpResult, VectorType]
+
+    def verify_(self):
+        res_type = self.res.typ.element_type
+        res_shape = self.res.typ.get_shape()
+
+        if res_type != self.lhs.typ.element_type or res_type != self.rhs.typ.element_type or res_type != self.acc.typ.element_type:
```

Comment:
Maybe split this into single checks for each one, s.t. we can report better error messages?